### PR TITLE
Atom/shawtar/blender scene exporter

### PR DESCRIPTION
**Pull request by HogJonny-AMZN**
Signed-off-by: shawstar <shawstar@amazon.com>

## What does this PR do?

O3DE Scene Exporter Update version 1.4

This add new features as well as bug fixes.
image

Version 1.4

In this version:

Add Custom Project Path: A tool to create a project path list that saves with your plugin for a quick way to add project asset paths and export assets.
User Defined (UDP) groundwork as well as Create PhysX Collider and LODs
Adding Animation Export Option for a Skin Attachment Mesh.

## How was this PR tested?

Tested in Blender 3.0+

### CODE REVIEW

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/constants.py [hunk 1]
```diff
@@ -20,13 +20,18 @@
 NO_ANIMATION = 'No Animation'
 KEY_FRAME_ANIMATION = 'Keyframe Animation'
 MESH_AND_RIG = 'Just Mesh with Rig'
+SKIN_ATTACHMENT = 'Skin Attachment Mesh with Rig'
 ANIMATION_LIST_OPTIONS = ( (
     ('0', NO_ANIMATION, 'Export with no keyframe Animation.'),
     ('1', KEY_FRAME_ANIMATION, 'Mesh needs to be parented to Armature with weights in order for O3DE to detect Entity as an Actor.'),
-    ('2', MESH_AND_RIG, 'Key All Bones, Force exporting at least one key of animation for all bones')
+    ('2', MESH_AND_RIG, 'Key All Bones, Force exporting at least one key of animation for all bones'),
+    ('3', SKIN_ATTACHMENT, 'Export a mesh with the Armature bones for use as a O3DE Skin Attachment.')
```
Comment (smurly) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/constants.py:28]: consider ending final lines in the collection with comma. python allows for trailing comma to reduce churn in future edits avoiding forgetting to add commas and having multiple lines change in diff.

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/constants.py [hunk 2]
```diff
@@ -20,13 +20,18 @@
...
+
+UPD = {'o3de_atom_lod' : '_lod',  'o3de_atom_phys' : '_phys'}
 TAG_O3DE = '.o3de'
 IMAGE_EXT = ('', '.jpg', '.png', '.JPG', '.PNG')
 USER_HOME = Path.home()
 DEFAULT_SDK_MANIFEST_PATH = Path.home().joinpath(f'{TAG_O3DE}','o3de_manifest.json')
-WIKI_URL = 'https://github.com/o3de/o3de/wiki/O3DE-DCCsi-Blender-Scene-Exporter'
\ No newline at end of file
+WIKI_URL = 'https://github.com/o3de/o3de/blob/development/Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/README.md'
+PLUGIN_VERSION = '1.4'
```
Comment (smurly) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/constants.py:37]: super trivial nit: this file is missing a final new line character.

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/o3de_utils.py [hunk 3]
```diff
@@ -24,22 +26,69 @@ def LookatEngineManifest():
...
     if engine_is_installed:
         # Make a list of projects to select from
         for project_full_path in o3de_projects:
-            list_o3de_projects.append((project_full_path, Path(project_full_path).name, project_full_path))
-    return list_o3de_projects
\ No newline at end of file
+            # Check to see if the project name might be 1 level up in this path if ending with project.
+            # For example we could have a path like this C:/Users/USERNAME/O3DE/Projects/loft-arch-vis-sample/Project
+            if Path(project_full_path).name == 'Project':
+                p = Path(project_full_path)
```
Comment (smurly) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/o3de_utils.py:77]: style: generally avoid using single letter variable names

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/o3de_utils.py [hunk 4]
```diff
@@ -24,22 +26,69 @@ def LookatEngineManifest():
...
+            o3de_projects = extend_project_list(o3de_projects)
             return o3de_projects, engine_is_installed
     except:
         o3de_projects = ['']
         engine_is_installed = False
         return o3de_projects, engine_is_installed
 
-def build_projects_list():
+def load_saved_projects():
+    """!
```
Comment (smurly) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/o3de_utils.py:37]: I'm not sure what the convention being used is with doc string quote `"""!` with the exclamation point

**Review by smurly — APPROVED** [review]

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/o3de_utils.py (hunk 4 above)
Comment (shawstar) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/o3de_utils.py:37]: We use doxygen to generate documentation. This is in line with the doxygen style guide.

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/constants.py (hunk 1 above)
Comment (shawstar) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/constants.py:28]: Thanks! That's a good power tip. Done.

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/o3de_utils.py (hunk 3 above)
Comment (shawstar) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/o3de_utils.py:77]: Yeah.. that's a good tip, I am generally good at not doing this, I must have missed it. Done

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/projects.json [hunk 5]
```diff
@@ -0,0 +1 @@
+["C:\\Users\\shawstar\\Documents\\3ds Max 2020", "C:\\Users\\shawstar\\Music"]
```
Comment (jeremyong-az) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/projects.json:1]: FYI hardcoded paths

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/ui.py [hunk 6]
```diff
@@ -46,6 +108,72 @@ def execute(self, context):
...
+    bl_idname = "vin.collider"
+    bl_label = "Add a PhysX Collider mesh to selected."
+
+    def execute(self, context):
+        """!
+        This function will create the collision mesh.
+        """
+        # Create a Pop-Up confirm window
+        bpy.types.Scene.pop_up_confirm_label = 'Adding UDP Type: o3de_atom_phys to mesh.'
+        bpy.types.Scene.pop_up_question_label = 'Add UPD to a Duplicate mesh?'
```
Comment (jeremyong-az) [inline Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/ui.py:154]: UPD typo

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/projects.json (hunk 5 above)
Comment (shawstar) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/projects.json:1]: Fixed

File: Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/ui.py (hunk 6 above)
Comment (shawstar) [reply Gems/AtomLyIntegration/TechnicalArt/DccScriptingInterface/Tools/DCC/Blender/AddOns/SceneExporter/ui.py]: Fixed

**Review by jeremyong-az — APPROVED** [review]

**Pull request merged**